src/openrct2/windows/_legacy.cpp

```diff
@@ -142,8 +142,8 @@ static std::tuple<bool, uint8_t> window_ride_construction_update_state_get_track
         endBank = _previousTrackBankEnd;
     }
 
-    uint16_t curve = _currentTrackCurve;
-    if (curve == 0xFFFF)
+    int32_t curve = _currentTrackCurve;
```

Comment:
auto

Reply:
These changes to `_currentTrackCurve` do not look correct. There is a lot of special code related to `_currentTrackCurve` and this needs some very very careful work to bring it up to `track_type_t`